monkey/infection_monkey/monkey.py

```diff
@@ -348,7 +354,10 @@ def _build_puppet(self) -> IPuppet:
             self._agent_event_publisher,
             self._propagation_credentials_repository,
         )
-        puppet = Puppet(self._agent_event_queue, plugin_registry)
+        plugin_compatability_verifier = PluginCompatabilityVerifier(
+            self._island_api_client, deepcopy(HARD_CODED_EXPLOITER_MANIFESTS)
```

Comment:
The `PluginCompatabilityVerifier` should perform the deepcopy operation. Doing it this way is leaking implementation details outside of the `PluginCompatabilityVerifier` component.

Reply:
See 6a43fbb64 and rebase on 2788-agent-os-compatibility-checks